pkg/yurthub/storage/disk/key_test.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2022 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package disk
+
+import (
+	"os"
+	"testing"
+
+	"github.com/openyurtio/openyurt/pkg/yurthub/storage"
+)
+
+var keyFuncTestDir = "/tmp/oy-diskstore-keyfunc"
+
+func TestKeyFunc(t *testing.T) {
+	cases := map[string]struct {
+		info   storage.KeyBuildInfo
+		key    string
+		err    error
+		isRoot bool
+	}{
+		"namespaced resource key": {
+			info: storage.KeyBuildInfo{
+				Component: "kubelet",
+				Resources: "pods",
+				Namespace: "kube-system",
+				Name:      "kube-proxy-xx",
+			},
+			key:    "kubelet/pods/kube-system/kube-proxy-xx",
+			isRoot: false,
+		},
+		"non-namespaced resource key": {
+			info: storage.KeyBuildInfo{
+				Component: "kubelet",
+				Resources: "nodes",
+				Name:      "edge-worker",
+			},
+			key:    "kubelet/nodes/edge-worker",
+			isRoot: false,
+		},
+		"resource list key": {
+			info: storage.KeyBuildInfo{
+				Component: "kubelet",
+				Resources: "pods",
+			},
+			key:    "kubelet/pods",
+			isRoot: true,
+		},
+		"resource list namespace key": {
+			info: storage.KeyBuildInfo{
+				Component: "kube-proxy",
+				Resources: "services",
+				Namespace: "default",
+			},
+			key:    "kube-proxy/services/default",
+			isRoot: true,
+		},
+		"no component err key": {
+			info: storage.KeyBuildInfo{
+				Resources: "nodes",
+			},
+			err: storage.ErrEmptyComponent,
+		},
+		"no resource err key": {
+			info: storage.KeyBuildInfo{
+				Component: "kubelet",
+				Name:      "edge-worker",
+			},
+			err: storage.ErrEmptyResource,
+		},
+	}
+
+	disk, err := NewDiskStorage(keyFuncTestDir)
+	if err != nil {
+		t.Errorf("failed to create disk store, %v", err)
+		return
+	}
+	keyFunc := disk.KeyFunc
+	for c, s := range cases {
+		key, err := keyFunc(s.info)
```

Comment:
you can add all the test contents in the following snippet:
```
t.Run(k, func(t *testing.T){
     xxxx
})
```

Reply:
@rambohe-ch Thanks, I'll fix it.

I'm now working on the autonomy test in the actual environment, and stucked at some problems. I'll create a issue talking about it later. So I think, this pr is not ready for review.

I'll remove the WIP, if I finish all the job.

Reply:
fixed